dev/io.openliberty.checkpoint_fat/fat/src/io/openliberty/checkpoint/fat/ConcurrencyTest.java

```diff
@@ -0,0 +1,223 @@
+ *     IBM Corporation - initial API and implementation
+ *******************************************************************************/
+package io.openliberty.checkpoint.fat;
+
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import org.jboss.shrinkwrap.api.ShrinkWrap;
+import org.jboss.shrinkwrap.api.spec.WebArchive;
+import org.junit.After;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import com.ibm.websphere.simplicity.ShrinkHelper;
+import com.ibm.websphere.simplicity.ShrinkHelper.DeployOptions;
+import com.ibm.websphere.simplicity.config.ServerConfiguration;
+
+import componenttest.annotation.Server;
+import componenttest.annotation.SkipIfCheckpointNotSupported;
+import componenttest.custom.junit.runner.FATRunner;
+import componenttest.topology.impl.LibertyServer;
+import concurrentApp.ConcurrentApp;
+import concurrentApp.RepeatedTrigger;
+import io.openliberty.checkpoint.spi.CheckpointPhase;
+
+@RunWith(FATRunner.class)
+@SkipIfCheckpointNotSupported
+public class ConcurrencyTest {
+
+    public static final String WAR_APP_NAME = "concurrentApp";
+    public static final String SERVER_NAME = "checkpointConcurrency";
+    public static final int MARGIN_OF_ERROR = 2000;
+    public static final SimpleDateFormat formatter = new SimpleDateFormat("m/d/yy, h:m:s:S z");
+
+    @Server(SERVER_NAME)
+    public static LibertyServer server;
+
+    @BeforeClass
+    public static void exportWebApp() throws Exception {
+        WebArchive war = ShrinkWrap.create(WebArchive.class, WAR_APP_NAME + ".war")
+                        .addClass(ConcurrentApp.class)
+                        .addClass(RepeatedTrigger.class);
+        ShrinkHelper.exportAppToServer(server, war, DeployOptions.OVERWRITE);
+    }
+
+    private long getTimeElapsed(String log1, String log2) throws Exception {
+        String timestampStr1 = log1.substring(log1.indexOf("[") + 1, log1.indexOf("]"));
+        Date timestamp1 = formatter.parse(timestampStr1);
+        String timestampStr2 = log2.substring(log2.indexOf("[") + 1, log2.indexOf("]"));
+        Date timestamp2 = formatter.parse(timestampStr2);
+        return Math.abs(timestamp2.getTime() - timestamp1.getTime());
+    }
+
+    @Test
+    public void testSchduledTaskTenSec() throws Exception {
+        Integer scheduledDelay = 10000;
+
+        ServerConfiguration config = server.getServerConfiguration();
+        config.getVariables().getById("scheduledTime").setValue(scheduledDelay.toString());
+        config.getVariables().getById("repeatTrigger").setValue("false");
+        config.getVariables().getById("repeatManagedExec").setValue("false");
+        server.updateServerConfiguration(config);
+
+        server.setCheckpoint(CheckpointPhase.APPLICATIONS, false, (server) -> {
+            String result = server.waitForStringInLog("Thread scheduled at startup", 20000);
```

Comment:
The waitforStringInLog is called after the server is halted (due to the checkpoint) so the 20 sec wait is not necessary, athough it does not affect the correctness of the test. It just introduces and unnecessary 20 sec wait when the check will fail.